---
title: "42_kallisto_to_deseq_to_accessionIDs"
author: "Aidan Coyle"
date: "Last compiled on `r format(Sys.time(), '%Y-%m-%d')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Originally written 2021/03/19
Roberts Lab, UW-SAFS
Written for analysis of _Hematodinium_ differential gene expression

**IMPORTANT: Many code blocks have been set to eval = FALSE due to long runtimes. To re-run analysis, change those blocks to TRUE**

# Purpose:
This script utilizes a series of custom-built functions for the following: 
- running a Trinity matrix of kallisto transcript counts through DESeq2
- turning DESeq2 output into newline-separated file of UniProt accessions
- creating Venn diagrams of transcript IDs and accession IDs (not needed for future steps - can skip if not wanted)



```{r libraries, message = FALSE, warning=FALSE}
# Add all required libraries that are installed with install.packages() here
list.of.packages <- "tidyverse"
# Add all libraries that are installed using BiocManager here
bioconductor.packages <- c("apeglm", "DESeq2", "vsn")

# Install BiocManager if needed
if(!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")

# Get names of all required packages that aren't installed
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[, "Package"])]
new.bioc.packages <- bioconductor.packages[!(bioconductor.packages %in% installed.packages()[, "Package"])]
# Install all new packages
if(length(new.packages)) install.packages(new.packages)
if(length(new.bioc.packages)) BiocManager::install(new.bioc.packages)

# Load all required libraries
all.packages <- c(list.of.packages, bioconductor.packages)
lapply(all.packages, FUN = function(X) {
  do.call("require", list(X))
})

# Functions are defined in hematodinium_analysis_functions.R
source("hematodinium_analysis_functions.R")
```

# Stage 1: Analyzing Trinity matrix from Kallisto output with DESeq2

Almost all analysis takes place in function deseq_analysis(). Documentation in function file. If attempting to redo a specific graph or table, the file DESeq_demonstration.R may be helpful

These blocks take quite a long time to run, so **in the markdown file, I have set eval to FALSE**. I recommend only re-running blocks that you are certain you want to run - again, run times are quite lengthy.

For all code blocks, **the order of rows in temperature and day values in exp_design should match the order of columns of the `kallisto.isoforms.counts.matrix` input file from Trinity for the pairwise comparison**. As an example, here's what the `exp_design` table should look like for hypothetical libraries A (day 0, lowered temp), B (day 2, ambient temp), and C (day 17, elevated temp) if the `kallisto.isoforms.counts.matrix` file columns are ordered A, B, C:

```{r example_table}
exp_design <- data.frame(temp = factor(c("Low", "Amb", "Elev")),
                         day = factor(c(0, 2, 17)))

exp_design
```

As a further note, the pairwise comparisons performed here treat temperature as water temperature at time of sample, **not** treatment group. This means that Elevated Day 0 and Lowered Day 0 libraries are treated as **ambient-temperature libraries**, as samples were taken prior to any temperature changes. For all other samples, there are no such differences.

### Explanation of deseq_analysis():

A function to analyze Kallisto output with DESeq2. Will likely take quite long to run - there's some computationally-heavy steps involved

Takes 4 inputs

1. kallisto_path: path to matrix created by Trinity from Kallisto outputs

2. experiment_table: table with treatments in experiment. 

  + Order should match columns of matrix from Trinity

  + Example: Day and temperature data for libraries 118, 132, 178, 463, 481, 485 (in order)

  + exp_design <- data.frame(day = factor(c(0, 0, 0, 17, 17, 17)), temp = factor(c("Amb", "Amb", "Amb", "Amb", "Amb", "Amb")))

3. output_path: path to the output folder

4. variable: variable being examined. Used in creation of deseq object.

  + Should be 'day' or 'temp' (without quotes)


### Elevated Day 0 vs. Elevated Day 2, Individual Libraries Only
exp_design = day and temperature data for libraries 173, 072, 127, 272, 294, 280 (in order). 

```{r elev0vselev2, eval = FALSE}

exp_design <- data.frame(temp = factor(c("amb", "amb", "amb",
                                         "elev", "elev", "elev")),
                         day = factor(c(0, 0, 0,
                                        2, 2, 2)))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/cbai_transcriptomev4.0/elev0_vs_elev2_indiv/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/cbai_transcriptomev4.0/elev0_vs_elev2_indiv",
               variable = "temp")
```
  



### Ambient Day 2 vs. Elevated Day 2, Individual Libraries Only
exp_design = day and temperature data for libraries 359, 349, 334, 272, 294, 280 (in order)


```{r amb2vselev2, eval = FALSE}

exp_design <- data.frame(temp = factor(c("amb", "amb", "amb",
                                         "elev", "elev", "elev")),
                         day = factor(c(2, 2, 2,
                                        2, 2, 2)))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/cbai_transcriptomev4.0/amb2_vs_elev2_indiv/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/cbai_transcriptomev4.0/amb2_vs_elev2_indiv",
               variable = "temp")
```

# Stage 2: Turning Transcript IDs Into Accession IDs

The previous step produced four tables for each pairwise comparison, as follows. Each table has the same columns: transcript ID, base mean, log2 fold change, lfcSE, stat, p-value, p-adjusted 

1. Data for all differentially-expressed transcripts (DEGlist.txt)

2. Same as 1 but with headers (DEGlist_wcols.txt)

3. Data for all genes (AllGenes.txt)

4. Same as 3, but with headers (AllGenes_wcols.txt)

This  step turns those files (specifically 2 and 4) and turns them into a newline-separated file of UniProt accession IDs by cross-referencing with the BLASTx table for the transcriptome used as an index for kallisto.

### Explanation of transcripts_to_geneIDs:

Turns DESeq2 output to UniProt accessions

Takes three inputs:

1. deseq_filepath: leads to file containing gene list from DESeq2

2. blast_filepath: path that leads to transcript ID/gene ID table

3. output_path: path to new newline-separated file of UniProt accession IDs

### UniProt accession IDs for differentially-expressed transcripts

```{r allDEGs}

# Elevated Day 0 vs. Elevated Day 2
transcripts_to_geneIDs(deseq_filepath =  "../graphs/DESeq2_output/cbai_transcriptomev4.0/elev0_vs_elev2_indiv/DEGlist_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/cbai4.0_blastxres.tab",
                       output_path =   "../output/accession_n_GOids/cbai_transcriptomev4.0/DEG_IDs/elev0_vs_elev2_indiv_DEG_IDs.txt")
# Ambient Day 2 vs. Elevated Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/cbai_transcriptomev4.0/amb2_vs_elev2_indiv/DEGlist_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/cbai4.0_blastxres.tab",
                       output_path = "../output/accession_n_GOids/cbai_transcriptomev4.0/DEG_IDs/amb2_vs_elev2_indiv_DEG_IDs.txt")
```

### UniProt accession IDs for all transcripts

```{r allgenes}
# Elevated Day 0 vs. Elevated Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/cbai_transcriptomev4.0/elev0_vs_elev2_indiv/AllGenes_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/cbai4.0_blastxres.tab",
                       output_path = "../output/accession_n_GOids/cbai_transcriptomev4.0/allgenes_IDs/elev0_vs_elev2_indiv_All_GeneIDs.txt")

# Ambient Day 2 vs. Elevated Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/cbai_transcriptomev4.0/amb2_vs_elev2_indiv/AllGenes_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/cbai4.0_blastxres.tab",
                       output_path = "../output/accession_n_GOids/cbai_transcriptomev4.0/allgenes_IDs/amb2_vs_elev2_indiv_All_GeneIDs.txt")
```


